src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,332 @@
+            location, version = _load_data(json_file.read())
+        return location, version
+
+    except Exception as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'An error occurred while trying to fetch driver details.') from e
+
+
+def _install_daqmx_driver(download_url: str) -> None:
+    """
+    Download and launch NI-DAQmx Driver installation in an interactive mode
+    """
+    temp_file = None
+    try:
+        with multi_access_temp_file() as temp_file:
+            logging.info("Downloading Driver to %s", temp_file)
+            urllib.request.urlretrieve(download_url, temp_file)
+            logging.info("Installing NI-DAQmx")
+            subprocess.run([temp_file], shell=True, check=True)
+    except subprocess.CalledProcessError as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Error occurred while installing NI-DAQmx driver. Command returned non-zero exit status.') from e
+    except urllib.error.URLError as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Error occurred while downloading NI-DAQmx driver') from e
+    except Exception as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Error occurred while installing NI-DAQmx driver') from e
+
+
+def _ask_user_confirmation(user_message: str) -> bool:
+    """
+    Prompt for user confirmation
+    This should be prompt or a default value and press enter? What should be user experience 
+
+    Args:
+        user_message (str): The message to display to the user.
+
+    Returns:
+        bool: True if the user confirms, False otherwise.
+
+    """
+    while True:
+        response = input(user_message + " (yes/no): ").strip().lower()
+        if response in ["yes", "y"]:
+            return True
+        elif response in ["no", "n"]:
+            return False
+        else:
+            print("Please enter 'yes' or 'no'.")
+
+
+def _create_user_question(latest_version: str, installed_version: str ) -> str:
+    """
+    Create a user question string for upgrading NI-DAQmx.
+    Args:
+        installed_version (str): The installed version of NI-DAQmx.
+        latest_version (str): The latest version of NI-DAQmx available.
+
+    Returns:
+        str: A string representing the user question.
+
+    >>> _create_user_question("24.0.0", "23.8.0")
+    'Installed NI-DAQmx version is 23.8.0. Latest version available is 24.0.0. Do you want to upgrade?'
+    """
+    user_question = (
+                      "Installed NI-DAQmx version is "
+                        + installed_version
+                        + ". Latest version available is "
+                        + latest_version
+                        + ". Do you want to upgrade?"
```

Comment:
fstring is cleaner

Reply:
Also please run Black on this file, just once?

Reply:
ok